sys/stat.go

```diff
@@ -0,0 +1,97 @@
+package sys
+
+import "io/fs"
+
+// Ino is the file serial number, or zero if unknown.
+//
+// Any constant value will invalidate functions that use Ino for equivalence,
+// such as os.SameFile.
+//
+// When zero is returned by a `readdir`, some compilers will attempt to
+// get a non-zero value with `lstat`. Those using Ino for darwin's definition
+// of `getdirentries` conflate zero `d_fileno` with a deleted file and skip the
+// entry. See /RATIONALE.md for more on this.
+type Ino = uint64
```

Comment:
I like that this is an alias, it gives room for documentation without really changing the semantics of the `Stat_t.Ino` field in relation to the other fields of the struct.